### PR TITLE
feat: [PAR-4866] add toggle for displaying product protection offer in cart

### PR DESCRIPTION
**Pull request by milesmcleod**
# Description

* toggle only displays in the admin panel if `Enable Product Protection (V2)` is set to `Yes` in the UI
* setting `Enable Product Protection (V2)` to `No` and saving config will automatically turn off `Display Product Protection Offers in Cart`
* Turning off `Enable Extend` will also automatically turn off `Display Product Protection Offers in Cart`
* This wil break any cypress tests that use the cart button (which might be...none of them?) when it is deployed to the automation stores because there will be no config value present and as such cart offers won't appear

## Ticket

https://helloextend.atlassian.net/browse/PAR-4249

## Type of change

Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Refactor (new code changes but everything functions the same)

## Checklist

- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have deployed and validated my code on a sandbox
- [ ] I have added tests that prove my fix is effective or that my feature works
 to

### CODE REVIEW

File: Model/Config/Backend/Enable.php [hunk 1]
```diff
@@ -52,6 +52,10 @@ public function afterSave()
         if ($isV2Enabled === 0) {
             $this->writer->save(\Extend\Integration\Service\Extend::ENABLE_PRODUCT_PROTECTION, 0);
             $this->writer->save(\Extend\Integration\Service\Extend::ENABLE_SHIPPING_PROTECTION, 0);
+            $this->writer->save(
```
Comment (milesmcleod) [inline Model/Config/Backend/Enable.php:55]: set to 0 when the whole extend integration is toggled off

File: Model/Config/Backend/EnableProductProtection.php [hunk 2]
```diff
@@ -43,7 +43,7 @@ public function __construct(
 
     /**
      * This will create a new Product Protection product when V2 PP is enabled
-     * If disabled, shipping protection and V2 product protection is disabled as well.
+     * If disabled, this will also disable cart offers
```
Comment (milesmcleod) [inline Model/Config/Backend/EnableProductProtection.php:46]: this comment was actually already out of date - just adding a note here for what it does now (when PP is turned off, disable cart offers)

File: etc/adminhtml/system.xml [hunk 3]
```diff
@@ -48,6 +48,16 @@
                     <frontend_model>Extend\Integration\Model\Config\Frontend\EnableProductProtection</frontend_model>
                     <backend_model>Extend\Integration\Model\Config\Backend\EnableProductProtection</backend_model>
                 </field>
+                <field id="enable_cart_offer" translate="label" type="select" sortOrder="20" showInDefault="1" canRestore="1">
+                    <label>Display Product Protection Offers in Cart</label>
+                    <source_model>Magento\Config\Model\Config\Source\Yesno</source_model>
+                    <depends>
+                        <field id="enable">1</field>
+                    </depends>
```
Comment (milesmcleod) [inline etc/adminhtml/system.xml:56]: the `<depends/>` element means that this field will only be visible if the neighboring `enable` toggle is true - i.e., if the user has elected to enable Product Protection. if Product Protection is disabled, this field will disappear, and will be saved as `No`/`0` when config is saved.

File: etc/config.xml [hunk 4]
```diff
@@ -13,6 +13,7 @@
             </shipping_protection>
             <product_protection>
                 <enable>0</enable>
+                <enable_cart_offer>0</enable_cart_offer>
```
Comment (milesmcleod) [inline etc/config.xml:16]: default config: cart offers do not display.

File: view/frontend/layout/checkout_cart_item_renderers.xml [hunk 5]
```diff
@@ -6,28 +6,28 @@
...
                 <arguments>
                     <argument name="viewModel" xsi:type="object">Extend\Integration\ViewModel\EnvironmentAndExtendStoreUuid</argument>
                 </arguments>
             </block>
         </referenceBlock>
         <referenceBlock name="checkout.cart.item.renderers.grouped.actions">
-            <block class="Magento\Checkout\Block\Cart\Item\Renderer\Actions\Generic" ifconfig="extend_plans/product_protection/enable" before="-" name="extend.cart.view.type.grouped.product-protection-offer" template="Extend_Integration::cart/item/renderer/actions/product-protection-simple-offer.phtml">
+            <block class="Magento\Checkout\Block\Cart\Item\Renderer\Actions\Generic" ifconfig="extend_plans/product_protection/enable_cart_offer" before="-" name="extend.cart.view.type.grouped.product-protection-offer" template="Extend_Integration::cart/item/renderer/actions/product-protection-simple-offer.phtml">
```
Comment (milesmcleod) [inline view/frontend/layout/checkout_cart_item_renderers.xml:30]: just changing every way that the cart offer is displayed to be conditional on the cart offer toggle (which is in turn conditional on the overall PP V2 toggle [which is in turn conditional on the `Enable Extend` toggle])

Comment (milesmcleod) [reply view/frontend/layout/checkout_cart_item_renderers.xml:30]: forgot to mention: unfortunately, `ifconfig` does not support multiple conditions.

File: etc/adminhtml/system.xml (hunk 3 above)
Comment (jomarsantos) [reply etc/adminhtml/system.xml:56]: There's no issue I'm guessing with the two fields above sharing the same `enable` id?

**Review by jomarsantos — APPROVED** [review]

File: etc/adminhtml/system.xml (hunk 3 above)
Comment (milesmcleod) [reply etc/adminhtml/system.xml:56]: yeah, i think it should be ok - i believe `depends` can only access sibling fields within a group, unless an absolute `<section>/<group>/<field>` path is passed to it

**Review by joshuathompsonextend — APPROVED** [review]

**Pull request merged**